etl/steps/data/meadow/hmd/2022-12-07/life_tables.py

```diff
@@ -0,0 +1,400 @@
+
+    Returns
+    -------
+    Dataset
+        Dataset with tables
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        # Decompress files
+        log.info("hmd_lt: decompressing files...")
+        decompress_file(local_file, tmp_dir)
+
+        # Check available files
+        log.info("hmd_lt: sanity checking folders...")
+        _sanity_check_relevant_folders(tmp_dir)
+
+        for folder in FOLDER_STRUCTURE:
+            log.info(f"hmd_lt: processing folder {folder['name']}...")
+            # Check that all required subolders are present in folder
+            path_folder = os.path.join(tmp_dir, folder["name"])
+            _sanity_check_files(path_folder, folder["files"])
+            # Load data
+            for f in folder["files"]:
+                log.info(f"hmd_lt: loading data from file {folder['name']}/{f}...")
+                path_subfolder = os.path.join(path_folder, f)
+                # Create table
+                table = make_table(path_subfolder, folder)
+                # add table to a dataset
+                log.info("hmd_lt: adding table to dataset...")
+                ds.add(table)
+    return ds
+
+
+def _sanity_check_relevant_folders(path: str) -> None:
+    folders = os.listdir(path)
+    assert all(
+        folder in folders for folder in FOLDERS_EXPECTED
+    ), f"Some of the life table ({FOLDERS_EXPECTED}) folders are not present!"
+
+
+def _sanity_check_files(path: str, files_expected: List[str]) -> None:
+    """Checks that all required files are present once zip is uncompressed."""
+    files_found = sorted(os.listdir(path))
+    assert (
+        files_found == files_expected
+    ), f"Files found are not the ones expected! Check that {files_expected} are actually there!"
+
+
+def make_table(input_folder: str, folder: dict) -> catalog.Table:
+    """Create table.
+
+    Loads data from `input_folder` and creates a table with the name `table_name`. It uses the metadata from `walden_ds`.
+
+    Parameters
+    ----------
+    input_folder : str
+        Folder containing uncompressed data from Walden.
+    folder : dict
+        Structure details from folder.
+
+    Returns
+    -------
+    catalog.Table
+        Table with data.
+    """
+    # List all files (each file corresponds to a country roughly)
+    files = glob(os.path.join(input_folder, "*.txt"))
+    log.info(f"hmd_lt: looking for available files in {input_folder}: {len(files)} found.")
+    # # Create df
+    age, year = _age_year(input_folder)
+    regex_header = folder["regex"].format(table=f"{age}x{year}")
+    df = make_df(files, regex_header)
+    # Clean df
+    df = clean_df(df)
+    # df to table
+    table = df_to_table(df, age, year, folder["sex"])
+    # # underscore all table columns
+    table = underscore_table(table)
+    return table
+
+
+def _age_year(input_folder: str) -> Tuple[str, str]:
+    ageyear = input_folder.split("_")[-1]
+    age, year = ageyear.split("x")
```

Comment:
You could easily check here that age and year are numbers.

Reply:
Thanks!